lms/djangoapps/courseware/courses.py

```diff
@@ -142,6 +142,19 @@ def check_course_access(course, user, action, check_if_enrolled=False):
             raise CourseAccessRedirect(reverse('about_course', args=[unicode(course.id)]))
 
 
+def can_user_enroll_in_course(course_key):
```

Comment:
After going through the `student/models.py` code, I'm not sure it makes sense to merge this functionality into existing code, but maybe we can move this into there and rename it `can_user_self_enroll_in_course` to make it clearer that instructors need to enroll the student in the course.